services/QuillLMS/app/models/activity_pack_sequence.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+# == Schema Information
+#
+# Table name: activity_pack_sequences
+#
+#  id             :bigint           not null, primary key
+#  release_method :string           default("immediate")
+#  created_at     :datetime         not null
+#  updated_at     :datetime         not null
+#  activity_id    :bigint
+#  classroom_id   :bigint
+#
+# Indexes
+#
+#  index_activity_pack_sequences_on_activity_id   (activity_id)
+#  index_activity_pack_sequences_on_classroom_id  (classroom_id)
+#
+# Foreign Keys
+#
+#  fk_rails_...  (activity_id => activities.id)
+#  fk_rails_...  (classroom_id => classrooms.id)
+#
+class ActivityPackSequence < ApplicationRecord
+  RELEASE_METHODS = [
+    IMMEDIATE_RELEASE = 'immediate',
+    STAGGERED_RELEASE = 'staggered'
+  ]
+
+  belongs_to :classroom
+  belongs_to :source, class_name: 'Activity', foreign_key: 'activity_id'
```

Comment:
I'd defer to everyone else's judgment on this, but I thought we were saying that the id on the table itself would be `source_id`, rather than the id being `activity_id` and just setting up `source` as the active record name. Having two different ways of calling the same thing here just seems a little ripe for confusion to me, but I'm the least versed in Rails architecture.

Reply:
I don't have strong feelings about what we use for `source`, but it definitely feels weird to preemptively de-couple the column name from the relation.  The current spec for the next round of scheduling-related stuff looks like it won't have a `source` at all, for instance.  It appears that the next thing we're likely to do i this space is put date-scheduling for assignment and due date on individual Activities in a single Activity Pack (unit).  I suppose that, in that universe, the `source` is a `unit_id` or something?  Might be worth deciding now if we think that the code in this PR is going to be reused in any way for the next scheduling feature set.

Reply:
After talking with @anathomical I'm going to go with diagnostic_activity_id since that's the current use case.   If at some point we add some more uses of ActivityPackSequence we can abstract then.